test/helpers/docker.go

```diff
@@ -0,0 +1,215 @@
+	log.Infof("Docker: set target to '%s'", target)
+	node := CreateNodeFromTarget(target)
+	if node == nil {
+		return nil
+	}
+
+	return &Docker{
+		Node:   node,
+		logCxt: log,
+	}
+}
+
+//ContainerExec executes cmd in the container with the provided name
+func (do *Docker) ContainerExec(name string, cmd string) *CmdRes {
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+
+	command := fmt.Sprintf("docker exec -i %s %s", name, cmd)
+	exit := do.Node.ExecWithSudo(command, stdout, stderr)
+	return &CmdRes{
+		cmd:    command,
+		stdout: stdout,
+		stderr: stderr,
+		exit:   exit,
+	}
+}
+
+//ContainerCreate runs an instance of the provided Docker image with network,
+//and with the provided options.
+func (do *Docker) ContainerCreate(name, image, net, options string) *CmdRes {
+
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+
+	cmd := fmt.Sprintf(
+		"docker run -d --name %s --net %s %s %s", name, net, options, image)
+	exit := do.Node.ExecWithSudo(cmd, stdout, stderr)
+
+	return &CmdRes{
+		cmd:    cmd,
+		stdout: stdout,
+		stderr: stderr,
+		exit:   exit,
+	}
+}
+
+//ContainerRm force a deletigion of a container based on a name
+func (do *Docker) ContainerRm(name string) *CmdRes {
+
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+
+	cmd := fmt.Sprintf("docker rm -f %s", name)
+	exit := do.Node.ExecWithSudo(cmd, stdout, stderr)
+
+	return &CmdRes{
+		cmd:    cmd,
+		stdout: stdout,
+		stderr: stderr,
+		exit:   exit,
+	}
+}
+
+//ContainerInspect runs docker inspect for the container with the provided name
+func (do *Docker) ContainerInspect(name string) *CmdRes {
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+
+	cmd := fmt.Sprintf("docker inspect %s", name)
+	exit := do.Node.ExecWithSudo(cmd, stdout, stderr)
+
+	return &CmdRes{
+		cmd:    cmd,
+		stdout: stdout,
+		stderr: stderr,
+		exit:   exit,
+	}
+}
+
+//ContainerInspectNet returns the networking information for the container with
+//the provided name
+func (do *Docker) ContainerInspectNet(name string) (map[string]string, error) {
+	res := do.ContainerInspect(name)
+	properties := map[string]string{
+		"EndpointID":        "EndpointID",
+		"GlobalIPv6Address": "IPv6",
+		"IPAddress":         "IPv4",
+		"NetworkID":         "NetworkID",
+	}
+
+	if !res.WasSuccessful() {
+		return nil, fmt.Errorf("Can not get the container %s", name)
+	}
+	filter := fmt.Sprintf(`{ [0].NetworkSettings.Networks.%s }`, networkName)
+	result := make(map[string]string)
+	data, err := res.FindResults(filter)
+	if err != nil {
+		return nil, err
+	}
+	for _, val := range data {
+		iface := val.Interface()
+		for k, v := range iface.(map[string]interface{}) {
+			if key, ok := properties[k]; ok {
+				result[key] = fmt.Sprintf("%s", v)
+			}
+		}
+	}
+	return result, nil
+}
+
+//NetworkCreate creates a Docker network of the provided name with the specified subnet
+func (do *Docker) NetworkCreate(name string, subnet string) *CmdRes {
```

Comment:
The caller of these `NetworkCreate()`, `NetworkDelete()`, `ContainerCreate()`, `ContainerRm()` functions never check the return values; Perhaps these should be modified to not return anything, then use ginkgo.Fail() if there's ever any error that occurs. This should ensure that we write our tests to properly instantiate and clean up containers and networks.

We may also consider just creating the docker network during `BeforeSuite()` if we just need it to exist for every test. That would simplify the tests.